sui_programmability/examples/deepbook/sources/custodian.move

```diff
@@ -0,0 +1,213 @@
+module deepbook::custodian {
+    use sui::balance::{Self, Balance, split};
+    use sui::coin::{Self, Coin};
+    use sui::object::{Self, UID, ID};
+    use sui::table::{Self, Table};
+    use sui::tx_context::TxContext;
+
+    friend deepbook::clob;
+
+    // Custodian for limit orders.
+
+    // <<<<<<<<<<<<<<<<<<<<<<<< Error codes <<<<<<<<<<<<<<<<<<<<<<<<
+    const EUserBalanceDoesNotExist: u64 = 1;
+    // <<<<<<<<<<<<<<<<<<<<<<<< Error codes <<<<<<<<<<<<<<<<<<<<<<<<
+
+    struct Account<phantom T> has store {
+        available_balance: Balance<T>,
+        locked_balance: Balance<T>,
+    }
+
+    struct AccountCap has key, store{ id: UID }
+
+    struct Custodian<phantom T> has key, store {
+        id: UID,
+        // Holds custody of protocol fees.
+        // The logic of withdrawing protocol fees is left out intentionally for further confirmation from Sui foundation.
+        custodian_balances: Balance<T>,
+        account_balances: Table<ID, Account<T>>,
+    }
+
+    public(friend) fun custodian_create_account<BaseAsset, QuoteAsset>(
+        base_custodian: &mut Custodian<BaseAsset>,
+        quote_custodian: &mut Custodian<QuoteAsset>,
+        ctx: &mut TxContext
+    ): AccountCap {
+        let accountCap = AccountCap { id: object::new(ctx) };
+        let user = get_account_cap_id(&accountCap);
+        table::add(
+                &mut base_custodian.account_balances,
+                user,
+                Account { available_balance: balance::zero(), locked_balance: balance::zero() }
+        );
+        table::add(
+                &mut quote_custodian.account_balances,
+                user,
+                Account { available_balance: balance::zero(), locked_balance: balance::zero() }
+            );
+        accountCap
+    }
+
+    public(friend) fun get_account_cap_id(account_cap: &AccountCap): ID {
```

Comment:
Should it be a friend? I can construct this ID manually and pass as an argument to a function.